gobblin-runtime/src/main/java/org/apache/gobblin/runtime/AbstractJobLauncher.java

```diff
@@ -412,6 +415,18 @@ public void apply(JobListener jobListener, JobContext jobContext)
         workUnitsCreationTimer.stop(this.eventMetadataGenerator.getMetadata(this.jobContext,
             EventName.WORK_UNITS_CREATION));
 
+        String workunitCreationGaugeName = MetricRegistry.name(
+            MetricReportUtils.GOBBLIN_JOB_METRICS_PREFIX,
+            TimingEvent.LauncherTimings.WORK_UNITS_CREATION, jobState.getJobName());
+        long workUnitsCreationTime = workUnitsCreationTimer.getDuration() / TimeUnit.SECONDS.toMillis(1);
+        ContextAwareGauge<Integer> workunitCreationGauge = RootMetricContext
+            .get().newContextAwareGauge(workunitCreationGaugeName, () -> (int) workUnitsCreationTime);
+        try {
+          RootMetricContext.get().register(workunitCreationGaugeName, workunitCreationGauge);
```

Comment:
I think registering the gauge with the RootMetricContext will not work in a multi threaded execution environment as in the case of gobblin cluster which would launch jobs in separate threads inside the same process. Can we create a new metric context (which will automatically become a child of the root metric context) and register the gauge with that context? We can assume that the metric reporters are recursive and will traverse the metric context tree and report all the gauges.